pyproject.toml

```diff
@@ -140,22 +140,49 @@ exclude=[
 ]
 
 [tool.ruff.lint]
+# This enables us to use CPY001: copyright header check
+preview = true
+# This enables us to use the explicit preview rules that we want only
+explicit-preview-rules = true
 # all rules can be found here: https://beta.ruff.rs/docs/rules/
-select = ["E", "F", "W", "I"]
+select = ["E", "F", "W", "I", "CPY001"]
 ignore=[
     # space before : (needed for how black formats slicing)
     "E203",
     # do not assign a lambda expression, use a def
     "E731",
     # do not use variables named 'l', 'O', or 'I'
     "E741",
+    # E721 is in preview (july 2024) and gives many false positives.
+    # Use `is` and `is not` for type comparisons, or `isinstance()` for
+    # isinstance checks
+    "E721",
+    # F841 is in preview (july 2024), and we don't care much about it.
+    # Local variable ... is assigned to but never used
+    "F841",
```

Comment:
These are things you noticed while working on this, not directly related to enabling the copyright check? Fine to have them, just for my education

Reply:
Yes, short answer: copyright check is in preview, we have to enable preview mode, that enables some unstable sub-rules that fail, to disable those sub-rules we have to disable the rule (xref: https://github.com/astral-sh/ruff/issues/12343)